katello/hooks/pre/30-upgrade.rb

```diff
@@ -37,6 +37,48 @@ def migrate_foreman
   Kafo::Helpers.execute('foreman-rake db:migrate')
 end
 
+def postgresql_10_upgrade
+  return unless facts[:os][:release][:major] == '7'
+
+  #Ensure there is at least 1x /var/lib/pgsql free disk space available on the main filesystem
+  current_postgres_dir = '/var/lib/pgsql'
+  new_postgres_dir = '/var/opt/rh/rh-postgresql10/lib/pgsql'
+
+  begin
+    until File.directory?(new_postgres_dir)
```

Comment:
The until loop is also in `available_space` so doesn't need to be duplicated here.